packages/ts/react-grid/test/grid-test-helpers.ts

```diff
@@ -29,6 +33,31 @@ export const getHeaderCell = <T>(grid: Grid<T>, row: number, col: number): HTMLE
 export const getHeaderCellContent = <T>(grid: Grid<T>, row: number, col: number): HTMLElement =>
   getCellContent(getHeaderCell(grid, row, col));
 
+export function getGrid(result: RenderResult): Grid {
+  return result.container.querySelector('vaadin-grid')!;
+}
+
+export async function nextFrame(): Promise<void> {
+  return new Promise((resolve) => {
+    requestAnimationFrame(() => {
+      resolve(undefined);
+    });
+  });
+}
+
+export async function setActiveItem<T>(grid: Grid<T>, item: T | undefined): Promise<void> {
+  grid.activeItem = item;
+  await nextFrame();
+  await nextFrame();
+  await nextFrame();
```

Comment:
What is the reason of `nextFrame` getting called here three times, and in many other places two time consecutively? What is it doing exactly?

Reply:
It is a bad way of waiting for React to finish rendering. It basically waits for the next frame in the browser, using `requestAnimationFrame`. Sometimes React renders in two cycles, sometimes in more. It needs to be fixed in a completely different way in the future

Reply:
Is it possible to introduce a function e.g. called `waitForRender` to wrap two or three `nextFrame()` calls? Would it break the tests if there is a need for two cycles and we wait for three cycles?

Reply:
Waiting more is just … waiting more. I would not wrap them though but instead use act() or waitFor()

Reply:
Created #1418 for this